src/components/stack-chart/index.js

```diff
@@ -83,32 +93,33 @@ class StackChartGraph extends React.PureComponent<Props> {
             </span>
           </div>
           <StackChartCanvas
-            // ChartViewport props
-            timeRange={timeRange}
-            maxViewportHeight={maxViewportHeight}
-            maximumZoom={this.getMaximumZoom()}
-            selection={selection}
-            updateProfileSelection={updateProfileSelection}
-            viewportNeedsUpdate={viewportNeedsUpdate}
-            // StackChartCanvas props
-            interval={interval}
-            thread={thread}
-            rangeStart={timeRange.start}
-            rangeEnd={timeRange.end}
-            stackTimingByDepth={stackTimingByDepth}
-            getCategory={getCategory}
-            getLabel={getLabel}
-            maxStackDepth={maxStackDepth}
-            stackFrameHeight={STACK_FRAME_HEIGHT}
+            viewportProps={{
+              timeRange: timeRange,
+              maxViewportHeight: maxViewportHeight,
+              maximumZoom: this.getMaximumZoom(),
+              selection: selection,
+              viewportNeedsUpdate: viewportNeedsUpdate,
+            }}
```

Comment:
nit: I think you can use the ES6 shorthand where possible, as that's just a JS object: `{{ timeRange, maxViewportHeight, ... }}`